docs/devel/consumers-claims.md

```diff
@@ -0,0 +1,297 @@
+  - [Status](#status)
+  - [General model](#general-model)
+  - [Storage](#storage)
+    - [PersistentVolumeClaim](#persistentvolumeclaim)
+    - [VolumeManagers](#volumemanagers)
+    - [Config](#config)
+    - [Scrubbing](#scrubbing)
+  - [Networking](#networking)
+    - [IngressClaim](#ingressclaim)
+    - [LoadBalancerManagers](#loadbalancermanagers)
+    - [Config](#config)
+    - [Scrubbing](#scrubbing)
+  - [Future](#future)
+
+<!-- END MUNGE: GENERATED_TOC -->
+
+## Background
+
+Some types of resources which are used in a cluster follow a common lifecycle:
+they are provisioned (whether manually or automatically), used by some
+consumer(s), eventually released by the consumer, scrubbed, and finally either
+released or reused.  These resources generally map to discrete physical or
+virtual _things_.
+
+Examples:
+   * Virtual storage (e.g. cloud-provider-hosted block devices)
+   * Physical storage (e.g. partitions on an iSCSI server or NFS shares)
+   * Virtual networking (e.g. cloud-provider-hosted load-balancers)
+   * Physical networking (e.g. VIPs on hardware load-balancers)
+
+These resources may have long lead times to provision (e.g. file a ticket to
+have an admin do work), may have requirements around de-provisioning (e.g.
+disks must be erased), or cost real money (e.g. for-pay cloud IaaS).  To use
+these things in Kubernetes we need a way to represent the in our API, to claim
+them as in-use, and to dynamically provision them (when supported).
+
+When thinking about these resources it is useful to keep in mind the extremes of
+how they might operate.  Consider:
+   * A cloud-based load-balancer which can be provisioned in seconds with just
+     an API call, but costs real money.
+   * A bare-metal disk device for which provisioning means sending a ticket to
+     your sys-admin, who eventually types some commands into a control
+     application and responds to your ticket with a device ID.
+
+## Status
+
+This document is somewhat aspirational.  The state of the system at the time of
+this writing is NOT fully aligned with this model, but we intend for the system
+to trend towards it.
+
+## General model
+
+Both storage and networking resources can be handled with a common model.
+
+```
+                       +----------+
+        +--create----> | resource |
+        |              +----------+
+   +----+----+               ^
+   | manager |---bind------> :
+   +----+----+               v
+        |              +----------+
+        +---watch----> |  claim   | <--+
+                       +----------+    |
+                             ^         |
+                             |         +---- user-created
+                             |         |
+                       +----------+    |
+                       | consumer | <--+
+                       +----------+
+```
+
+In this model a `consumer` object (created by a user in the API) files a
+`claim` (also created by the user) for some resource.  Claims are abstract by
```

Comment:
maybe "some type of resource" to make it clearer it's not a specific resource instance?